### PR TITLE
Expanded DOM Strategy

### PR DESCRIPTION
**Pull request by Innos**
expanded DOM strategy to work with async methods and ReactJS

### CODE REVIEW

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndCheckExecutionStrategy.cs [hunk 1]
```diff
@@ -183,16 +183,18 @@ public override ExecutionResult Execute(ExecutionContext executionContext)
             // In NodeJS there is no compilation
             result.IsCompiledSuccessfully = true;
 
+            var executor = new RestrictedProcessExecutor();
+
             // Preprocess the user submission
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndCheckExecutionStrategy.cs]: Do we need all these comments on every action? Code is quite self explanatory. @KristianMariyanov ?

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndCheckExecutionStrategy.cs [hunk 2]
```diff
@@ -251,7 +253,7 @@ protected virtual List<TestResult> ProcessTests(ExecutionContext executionContex
 
         protected string ProcessModulePath(string path) => path.Replace('\\', '/');
 
-        protected virtual string PreprocessJsSubmission(string template, ExecutionContext context)
+        protected virtual string PreprocessJsSubmission(string template, ExecutionContext context,IExecutor executor)
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndCheckExecutionStrategy.cs]: Please, space after the comma

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndCheckExecutionStrategy.cs [hunk 3]
```diff
@@ -251,7 +253,7 @@ protected virtual List<TestResult> ProcessTests(ExecutionContext executionContex
...
         {
             var problemSkeleton = context.TaskSkeletonAsString ??
                 "function adapter(input,code) {return code(input);}";
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndCheckExecutionStrategy.cs]: space after comma; space before/after curly brackets (from the inner side)

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndCheckExecutionStrategy.cs (hunk 1 above)
Comment (KristianMariyanov) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndCheckExecutionStrategy.cs]: This specific is redundant IMO.

Comment (Innos) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndCheckExecutionStrategy.cs]: The comments may be mostly redundant, but in the base classes they are usefull to orient new people who may wish to work on the code.

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy.cs [hunk 4]
```diff
@@ -1,4 +1,6 @@
-﻿namespace OJS.Workers.ExecutionStrategies
+﻿using OJS.Common.Extensions;
```
Comment (KristianMariyanov) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy.cs]: Plase put all usings inside namespaces

**Pull request merged**